code/model/editableformfields/EditableFileField.php

```diff
@@ -43,30 +46,58 @@ public function getCMSFields()
         );
 
         $fields->addFieldToTab("Root.Main", new LiteralField("FileUploadWarning",
-                "<p class=\"message notice\">" . _t("UserDefinedForm.FileUploadWarning",
+            "<p class=\"message notice\">" . _t("UserDefinedForm.FileUploadWarning",
                 "Files uploaded through this field could be publicly accessible if the exact URL is known")
-                . "</p>"), "Type");
+            . "</p>"), "Type");
+
+        $fields->addFieldToTab(
+            'Root.Main',
+            NumericField::create('MaxFileSizeMB')
+                ->setTitle('Max File Size MB')
+                ->setDescription("Note: Maximum php allowed size is {$this->getPHPMaxFileSize()} MB")
+        );
 
         return $fields;
     }
 
+    /**
+     * @return ValidationResult
+     */
+    public function validate()
+    {
+        $result = parent::validate();
+
+        if ($this->MaxFileSizeMB > self::get_php_max_file_size()) {
+            $max = self::get_php_max_file_size();
```

Comment:
It's a general principle, applied to this code because that's the code I was looking at where I saw it. I don't think there's a significant probability that it will need to be overridden.

But let's assume someone wants to get this module ready for 4.0 (currently alpha), where the references to `File::ini2bytes` references an invalid namespace. Or we could imagine a scenario where we want to check not only for the maximum allowed file upload size for PHP (on the local server) but also check for limits on a third-party server where the file will subsequently be uploaded...

The point isn't "what reasons can we think of that people might need to override?" but rather "if we allow people to override (read: subclass) then this works against that"...